torch/csrc/jit/codegen/cuda/lower_predicate.cpp

```diff
@@ -100,6 +100,22 @@ class ConditionalFromPredicateModifier {
             pred->thread_pred(),
             pred->predicate_type());
       }
+      case PredicateType::Vectorize: {
+        std::vector<kir::ForLoop*> outer_loops;
+        kir::ForLoop* vectorized_loop = nullptr;
+        for (auto loop : for_loops_structure_) {
+          if (loop->iter_domain()->parallelType() == ParallelType::Vectorize) {
+            vectorized_loop = loop;
+            break;
+          } else {
+            outer_loops.emplace_back(loop);
+          }
+        }
+        TORCH_INTERNAL_ASSERT(
+            vectorized_loop != nullptr, "Should be unreachable.");
+        return UnswitchPredicate::get(
+            outer_loops, vectorized_loop, p2c_root_map_);
+      }
```

Comment:
I understand how the other parts of this PR fixes the problem, but I'm not sure about this part. Is this also necessary? Or, is this just refactoring?

Reply:
It is necessary because `outer_loops != for_loop_structure_` so the code can't be shared with `PredicateType::Unswitch`.

Reply:
It is also a small refactor since I moved this code from `lower_unroll.cpp`

Reply:
Ah, okay, makes sense.